core/include/sched.h

```diff
@@ -172,6 +172,10 @@ typedef enum {
     STATUS_FLAG_BLOCKED_ALL,        /**< waiting for all flags in flag_mask       */
     STATUS_MBOX_BLOCKED,            /**< waiting for get/put on mbox              */
     STATUS_COND_BLOCKED,            /**< waiting for a condition variable         */
+#if IS_USED(MODULE_ESP_FREERTOS_COMMON)
+    STATUS_QUEUE_FULL_BLOCKED,      /**< FreeRTOS compatibility layer queue full  */
+    STATUS_QUEUE_EMPTY_BLOCKED,     /**< FreeRTOS compatibility layer queue empty */
+#endif
```

Comment:
If you add those at the end it will not mess up the thread state mapping in OpenOCD.

The use of this enum in OpenOCD also would speak for adding this unconditionally. 
We already have `STATUS_ZOMBIE` which is only used by `mynewt-core`

Reply:
> If you add those at the end it will not mess up the thread state mapping in OpenOCD.
> 
> The use of this enum in OpenOCD also would speak for adding this unconditionally. We already have `STATUS_ZOMBIE` which is only used by `mynewt-core`

Hm, I got an assert here https://github.com/RIOT-OS/RIOT/blob/25a52691508c4547c1998fc6fb994f29b97652e0/core/thread.c#L156 when I had them at the end.

Reply:
> We already have `STATUS_ZOMBIE` which is only used by `mynewt-core`

I couldn't find that.

Reply:
It's set by `thread_zombify()` which is only called by that package.

I guess the assert could be replaced by 

```
    assert(thread->status == STATUS_RUNNING || thread->status == STATUS_PENDING);
```

but I'm not sure if this assumption is backed in elsewhere too :confused:

Reply:
According to the documentation, `thread_add_to_list` may only be called for threads that are not in a _runqueue_. This is ensured by the `assert`.

Reply:
> It's set by `thread_zombify()` which is only called by that package.
> 
> I guess the assert could be replaced by
> 
> ```c
> assert(thread->status == STATUS_RUNNING || thread->status == STATUS_PENDING);
> ```
> 
> but I'm not sure if this assumption is backed in elsewhere too confused

Ah ok, it is a permanent state.

Reply:
> I guess the assert could be replaced by
> 
> ```c
> assert(thread->status == STATUS_RUNNING || thread->status == STATUS_PENDING);
> ```
> 
> but I'm not sure if this assumption is backed in elsewhere too confused

Hm, ` >= STATUS_ON_RUNQUEUE` is also used several times:
```
263:        if (!(process->status >= STATUS_ON_RUNQUEUE)) {
268:        if (process->status >= STATUS_ON_RUNQUEUE) {
280:    int on_runqueue = (active_thread->status >= STATUS_ON_RUNQUEUE);
./core/sched.c
146:    if (me->status >= STATUS_ON_RUNQUEUE) {
./core/thread.c
```

Reply:
I see, it seems not to be a short-term solution. Maybe, we should continue with PR #18439 with state `STATUS_MUTEX_BLOCKED` for now which also solves the problem.

Reply:
Defining them unconditionally would cost 18 additional bytes in ROM.